cmd/atlas/internal/cmdapi/schema.go

```diff
@@ -279,6 +305,68 @@ func CmdApplyRun(cmd *cobra.Command, _ []string) error {
 	return applyRun(cmd, c, ApplyFlags.DevURL, ApplyFlags.Paths, ApplyFlags.DryRun, ApplyFlags.AutoApprove, GlobalFlags.Vars)
 }
 
+// CmdCleanRun is the command executed when running the CLI with 'schema clean' args.
+func CmdCleanRun(cmd *cobra.Command, _ []string) error {
+	// Open a client to the database.
+	c, err := sqlclient.Open(cmd.Context(), CleanFlags.URL)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+	// Acquire a lock.
+	if l, ok := c.Driver.(schema.Locker); ok {
+		unlock, err := l.Lock(cmd.Context(), "atlas_migrate_execute", 0)
+		if err != nil {
+			return fmt.Errorf("acquiring database lock: %w", err)
+		}
+		// If unlocking fails notify the user about it.
+		defer cobra.CheckErr(unlock())
+	}
+	var drop []schema.Change
+	// If the connection is bound to a schema, only drop the resources inside the schema.
+	switch c.URL.Schema {
+	case "":
+		r, err := c.InspectRealm(cmd.Context(), nil)
+		if err != nil {
+			return err
+		}
+		drop, err = c.RealmDiff(r, nil)
+		if err != nil {
+			return err
+		}
+	default:
+		s, err := c.InspectSchema(cmd.Context(), c.URL.Schema, nil)
+		if err != nil {
+			return err
+		}
+		drop, err = c.SchemaDiff(s, schema.New(s.Name))
+		if err != nil {
+			return err
+		}
+	}
+	if len(drop) == 0 {
+		cmd.Println("Nothing to drop")
+		return nil
+	}
+	p, err := c.PlanChanges(cmd.Context(), "", drop)
+	if err != nil {
+		return err
+	}
+	cmd.Println("-- Planned Changes:")
+	for _, c := range p.Changes {
+		if c.Comment != "" {
+			cmd.Println("--", strings.ToUpper(c.Comment[:1])+c.Comment[1:])
+		}
+		cmd.Println(c.Cmd)
+	}
+	if CleanFlags.AutoApprove || promptUser() {
+		if err := c.ApplyChanges(cmd.Context(), drop); err != nil {
+			return err
+		}
+	}
```

Comment:
I don't know how much it's needed, but some stuff here can be moved out to shared functions on the same file and used by the 2 commands (apply and clean).